server/src/main/java/org/apache/iotdb/db/engine/cache/BloomFilterCache.java

```diff
@@ -0,0 +1,183 @@
+      config.getAllocateMemoryForBloomFilterCache();
+  private static final boolean CACHE_ENABLE = config.isMetaDataCacheEnable();
+
+  private final LoadingCache<BloomFilterCacheKey, BloomFilter> lruCache;
+
+  private BloomFilterCache() {
+    if (CACHE_ENABLE) {
+      logger.info("BloomFilterCache size = " + MEMORY_THRESHOLD_IN_BLOOM_FILTER_CACHE);
+    }
+    lruCache =
+        Caffeine.newBuilder()
+            .maximumWeight(MEMORY_THRESHOLD_IN_BLOOM_FILTER_CACHE)
+            .weigher(
+                (Weigher<BloomFilterCacheKey, BloomFilter>)
+                    (key, bloomFilter) ->
+                        (int)
+                            (RamUsageEstimator.shallowSizeOf(key)
+                                + RamUsageEstimator.sizeOf(key.tsFilePrefixPath)
+                                + RamUsageEstimator.sizeOf(bloomFilter)))
+            .recordStats()
+            .build(
+                key -> {
+                  try {
+                    TsFileSequenceReader reader =
+                        FileReaderManager.getInstance().get(key.filePath, true);
+                    return reader.readBloomFilter();
+                  } catch (IOException e) {
+                    logger.error(
+                        "Something wrong happened in reading bloom filter in tsfile {}",
+                        key.filePath,
+                        e);
+                    throw e;
+                  }
+                });
+  }
+
+  public static BloomFilterCache getInstance() {
+    return BloomFilterCacheHolder.INSTANCE;
+  }
+
+  public BloomFilter get(BloomFilterCacheKey key) throws IOException {
+    return get(key, false);
+  }
+
+  public BloomFilter get(BloomFilterCacheKey key, boolean debug) throws IOException {
+    if (!CACHE_ENABLE) {
+      TsFileSequenceReader reader = FileReaderManager.getInstance().get(key.filePath, true);
+      return reader.readBloomFilter();
+    }
+
+    BloomFilter bloomFilter = lruCache.get(key);
+
+    if (debug) {
+      DEBUG_LOGGER.info("get bloomFilter from cache where filePath is: " + key.filePath);
+    }
+
+    return bloomFilter;
+  }
+
+  public double calculateChunkHitRatio() {
+    return lruCache.stats().hitRate();
+  }
+
+  public long getEvictionCount() {
+    return lruCache.stats().evictionCount();
+  }
+
+  public long getMaxMemory() {
+    return MEMORY_THRESHOLD_IN_BLOOM_FILTER_CACHE;
+  }
+
+  public double getAverageLoadPenalty() {
+    return lruCache.stats().averageLoadPenalty();
+  }
+
+  /** clear LRUCache. */
+  public void clear() {
+    lruCache.invalidateAll();
+    lruCache.cleanUp();
+  }
+
+  public void remove(BloomFilterCacheKey key) {
+    lruCache.invalidate(key);
+  }
+
+  @TestOnly
+  public BloomFilter getIfPresent(BloomFilterCacheKey key) {
+    return lruCache.getIfPresent(key);
+  }
+
+  @TestOnly
+  public void clearUp() {
+    lruCache.cleanUp();
+  }
+
+  public static class BloomFilterCacheKey {
+
+    private final String filePath;
```

Comment:
```suggestion
    // There is no need to add this field size while calculating the size of BloomFilterCacheKey, because filePath is get from TsFileResource, different BloomFilterCacheKey of the same file share this String.
    private final String filePath;
```